src/components/MultiHopTrade/types.ts

```diff
@@ -7,3 +11,41 @@ export type StepperStep = {
   content?: JSX.Element
   status?: MultiHopExecutionStatus
 }
+
+export enum SelectedQuoteStatus {
+  ReadyToPreview = 'ReadyToPreview',
+  Loading = 'Loading',
+  Updating = 'Updating',
+  SellAmountBelowMinimum = 'SellAmountBelowMinimum',
+  SellAmountBelowTradeFee = 'SellAmountBelowTradeFee',
+  InsufficientFirstHopFeeAssetBalance = 'InsufficientFirstHopFeeAssetBalance',
+  InsufficientLastHopFeeAssetBalance = 'InsufficientLastHopFeeAssetBalance',
+  InsufficientSellAssetBalance = 'InsufficientSellAssetBalance',
+  NoConnectedWallet = 'NoConnectedWallet',
+  SellAssetNotNotSupportedByWallet = 'SellAssetNotNotSupportedByWallet',
+  BuyAssetNotNotSupportedByWallet = 'BuyAssetNotNotSupportedByWallet',
+  NoReceiveAddress = 'NoReceiveAddress',
+  NoQuotesAvailableForTradePair = 'NoQuotesAvailableForTradePair',
+  NoQuotesAvailableForSellAmount = 'NoQuotesAvailableForSellAmount',
+  TradingInactiveOnSellChain = 'TradingInactiveOnSellChain',
+  TradingInactiveOnBuyChain = 'TradingInactiveOnBuyChain',
+  NoQuotesAvailable = 'NoQuotesAvailable',
+  UnknownError = 'UnknownError',
+}
+
+export type QuoteStatus = {
+  selectedQuoteErrors: SelectedQuoteStatus[]
+  quoteStatusTranslation: string | [string, InterpolationOptions]
+  quoteHasError: boolean
+  errorMessage?: string
```

Comment:
is the error message user facing?

also can we combine `selectedQuoteErrors`,  `errorMessage` and `quoteHasError` into 1 and infer the uses of them using the one?

Reply:
Removed `quoteHasError` in https://github.com/shapeshift/web/pull/4763/commits/af67c60008afef05b420d549558393778762aff6.

I'll have more of a think about `selectedQuoteErrors` and `errorMessage`.

Reply:
Improved upon this by passing the full monadic error state down to the view layer: https://github.com/shapeshift/web/pull/4763/commits/3f1a0ea7fc7400da5cf1c2ed7517c806ab7ec2e9